From 6c4cd35a04c882a935ad167da92ae7397bfea345 Mon Sep 17 00:00:00 2001
From: Gabriele M <moto.falcon.git@gmail.com>
Date: Fri, 14 Apr 2017 11:34:19 +0200
Subject: [PATCH 5/6] Zygote: Fix GL preload property evaluation

Commit b3a1417663d64fe003b03c97e2f24b6 ("Zygote: Restore GL preload")
should allow to disable GL preloading on devices with no updatable
drivers support (i.e. empty ro.gfx.driver.0). However, the property
ro.zygote.disable_gl_preload has no effect on a device with no
updatable driver support and it does on a device with updatable
driver support, which is the opposite of what we want.

Change-Id: I3df6c929a465441632ca1540dbb575daa174b0fb
---
 core/java/com/android/internal/os/ZygoteInit.java | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/java/com/android/internal/os/ZygoteInit.java b/core/java/com/android/internal/os/ZygoteInit.java
index e4f573b..0d3f50c 100644
--- a/core/java/com/android/internal/os/ZygoteInit.java
+++ b/core/java/com/android/internal/os/ZygoteInit.java
@@ -245,8 +245,8 @@ public class ZygoteInit {
 
     private static void preloadOpenGL() {
         String driverPackageName = SystemProperties.get(PROPERTY_GFX_DRIVER);
-        if (!SystemProperties.getBoolean(PROPERTY_DISABLE_OPENGL_PRELOADING, false) ||
-                driverPackageName == null || driverPackageName.isEmpty()) {
+        if (!SystemProperties.getBoolean(PROPERTY_DISABLE_OPENGL_PRELOADING, false) &&
+                (driverPackageName == null || driverPackageName.isEmpty())) {
             EGL14.eglGetDisplay(EGL14.EGL_DEFAULT_DISPLAY);
         }
     }
-- 
2.7.4

